### PR TITLE
try to optimize hot spots for orc-booting compiler

### PR DESCRIPTION
**Pull request by ringabout**

### CODE REVIEW

File: compiler/astalgo.nim [hunk 1]
```diff
@@ -815,16 +815,21 @@ type
     name*: PIdent
 
 proc nextIdentIter*(ti: var TIdentIter, tab: TStrTable): PSym =
+  # hot spots
   var h = ti.h and high(tab.data)
   var start = h
-  result = tab.data[h]
-  while result != nil:
-    if result.name.id == ti.name.id: break
+  var p = addr tab.data[h]
```
Comment (Araq) [inline compiler/astalgo.nim]: No. Instead use a `.cursor`.

File: compiler/astalgo.nim [hunk 2]
```diff
@@ -815,16 +815,21 @@ type
...
+  while p != nil:
```
Comment (Araq) [inline compiler/astalgo.nim:825]: This would become `while p[] != nil:` with your solution.

File: compiler/astalgo.nim (hunk 1 above)
Comment (ringabout) [reply compiler/astalgo.nim]: I used `.cursor` https://github.com/nim-lang/Nim/pull/20001/commits/e686303cd3e82c71334791b446345925e065c03e however it cannot pass CI

https://github.com/nim-lang/Nim/runs/7277971587?check_suite_focus=true

```
 /home/runner/work/Nim/Nim/compiler/astalgo.nim(821, 11) Error: invalid pragma: cursor
```

This PR probably needs to wait for csource updates.

File: compiler/astalgo.nim (hunk 2 above)
Comment (ringabout) [reply compiler/astalgo.nim:825]: I see

File: compiler/astalgo.nim (hunk 1 above)
Comment (Araq) [reply compiler/astalgo.nim]: No, just emulate the pragma like so:

```nim

when not defined(nimHasCursor):
  {.pragma: cursor.}
```

Comment (ringabout) [reply compiler/astalgo.nim]: Thank you!

**Pull request merged**